code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -2500,3 +2502,66 @@
 	damage *= bonus
 	hit_and_run = 1 //reset the hit and run bonus
 	return damage
+
+/mob/living/carbon/Xenomorph/Hivelord/proc/hivelord_sting(mob/living/L, count = 2)
+
+	if(!check_state())
+		return
+
+	if(world.time < last_hivelord_sting + HIVELORD_STING_COOLDOWN) //Sure, let's use this.
+		to_chat(src, "<span class='xenodanger'>You are not ready to Sting again. It will be ready in [(last_hivelord_sting + HIVELORD_STING_COOLDOWN - world.time) * 0.1] seconds.</span>")
+		return
+
+	if(stagger)
+		to_chat(src, "<span class='xenowarning'>You try to sting but are too disoriented!</span>")
+		return
+
+	if(!L.can_zombie() )
+		to_chat(src, "<span class='xenowarning'>Your sting won't affect this corpse. A larva took the essence, it's too decayed, or simply too fresh.</span>")
+		return
+
+	if(!Adjacent(L))
+		if(world.time > (recent_notice + notice_delay)) //anti-notice spam
+			to_chat(src, "<span class='xenowarning'>You can't reach this target!</span>")
+			recent_notice = world.time //anti-notice spam
+		return
+
+	if(!check_plasma(150))
+		return
+	last_hivelord_sting = world.time
+	use_plasma(150)
+
+	addtimer(CALLBACK(src, .hivelord_sting_cooldown), HIVELORD_STING_COOLDOWN)
+
+	while(count)
+		face_atom(L)
+		if(!do_after(src, HIVELORD_STING_CHANNEL_TIME, TRUE, 5, BUSY_ICON_HOSTILE))
+			return
+		if(!Adjacent(L) || stagger)
+			return FALSE
+		animation_attack_on(L)
+		playsound(L, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
+		L.change_mob_type(/mob/living/carbon/Xenomorph/Necromorph)
+		L.gib()
+
+/mob/living/carbon/Xenomorph/Hivelord/proc/hivelord_sting_cooldown()
+	playsound(loc, 'sound/voice/alien_drool1.ogg', 50, 1)
+	to_chat(src, "<span class='xenodanger'>You feel your virus glands refil, another body can be prepared for reanimation.</span>")
+	update_action_button_icons()
+
+/mob/proc/can_zombie()
+	return FALSE
+
+/mob/living/carbon/monkey/can_zombie()
+	return FALSE
+
+/mob/living/carbon/human/can_zombie()
+	if(stat == DEAD && timeofdeath > 500 && !check_tod() && !chestburst) // Minimum amount of time required (500) so roundstart corpses are excluded from being able to be revived.
```

Comment:
There's better ways to check if a mob had a client inside. Next_move should be the safest (if it's not null, it was at one point controlled by a player).